src/extension.ts

```diff
@@ -42,6 +42,27 @@ export async function activate() {
         }
       });
 
+      // Register smart select command
+      commands.registerCommand(Constants.smartSelectCommand, (cell: NotebookCell) => {
+        const id = cell.notebook.uri.toString();
+        const provider = gatherProviderMap.get(id);
```

Comment:
I don't think that you have to change this now, but in Jupyter we use a WeakMap<NotebookDocument, X> for this type of thing (see something like NotebookIpyWidgetCoordinator). Helps a bit as you can just use the NotebookDocument as the key, but not keep a GC ref on it.

Reply:
Good call, I used the URI as a key instead of NotebookDocument to have it work in the interactive window case.